usr/lib/common/mech_aes.c

```diff
@@ -1191,6 +1270,135 @@ CK_RV aes_ctr_decrypt_update(STDLL_TokData_t *tokdata,
     }
 }
 
+static CK_RV aes_xts_crypt_update(STDLL_TokData_t *tokdata,
+                                  SESSION *sess,
+                                  CK_BBOOL length_only,
+                                  CK_BBOOL encrypt,
+                                  ENCR_DECR_CONTEXT *ctx,
+                                  CK_BYTE *in_data,
+                                  CK_ULONG in_data_len,
+                                  CK_BYTE *out_data, CK_ULONG *out_data_len)
+{
+    AES_XTS_CONTEXT *context = NULL;
+    OBJECT *key = NULL;
+    CK_BYTE *clear = NULL;
+    CK_ULONG total, remain, out_len;
+    CK_RV rc;
+
```

Comment:
For shorter path length we could also check here if (in_data == NULL || in_data_len == 0) and just return

Reply:
this has already been checked by the caller (i.e. in SC_EncryptUpdate)

Reply:
... and `in_data_len == 0` is actually allowed. Also `in_data` may be `NULL` if `in_data_len` is zero. 
Only if `in_data_len` is not zero, then `in_data` must also be non-NULL.